setup/src/setup-haskell.ts

```diff
@@ -21,34 +29,106 @@ export default async function run(
   try {
     core.info('Preparing to setup a Haskell environment');
     const os = process.platform as OS;
+    const home = process.env.HOME;
     const opts = getOpts(getDefaults(os), os, inputs);
 
+    // Cache parameters
+    let cachePaths: string[] = [];
+    let hashFile = '';
+
     for (const [t, {resolved}] of Object.entries(opts).filter(o => o[1].enable))
       await core.group(`Installing ${t} version ${resolved}`, async () =>
         installTool(t as Tool, resolved, os)
       );
 
-    if (opts.stack.setup)
+    if (opts.stack.setup) {
       await core.group('Pre-installing GHC with stack', async () =>
         exec('stack', ['setup', opts.ghc.resolved])
       );
+    }
 
-    if (opts.cabal.enable)
+    if (opts.stack.enable) {
+      const stackRoot = os === 'win32' ? 'C:\\sr' : `${home}/.stack`;
+      core.setOutput('stack-root', stackRoot);
+      if (os === 'win32') core.exportVariable('STACK_ROOT', 'C:\\sr');
+
+      if (opts.cache) {
```

Comment:
We will probably want to be fairly careful here. Is there a plan for how the caching behavior should work if the defaults aren't overridden vs if the defaults are overridden?

In particular, I'm not sure it's completely intuitive to have "enabling the cache" result in running commands or otherwise creating files/directories you might not otherwise expect.

I also don't actually know how globbing and then accessing the first item of the array works if there's no `stack.yaml.lock` file. Does it throw an error? Would the `**/stack.*.lock` potentially grab way too many files and/or the wrong file(s) if you had git submodules checked out and one of your submodules had a `stack.yaml.lock` file but your stack project did not?